frontend/packages/console-shared/src/components/dynamic-form/types.ts

```diff
@@ -7,3 +7,23 @@ export enum SchemaType {
   array = 'array',
   object = 'object',
 }
+
+export type DynamicFormFieldOptionsList = {
+  label: string;
+  value: string;
+}[];
+
+export type DynamicFormFieldDependency = {
+  controlFieldPath: string;
+  controlFieldValue: string;
+  controlFieldName: string;
+};
+
+export type UiSchemaOptionsWithDependency = {
```

Comment:
nit: console naming conventions would make this capital `UI`

```suggestion
export type UISchemaOptionsWithDependency = {
```